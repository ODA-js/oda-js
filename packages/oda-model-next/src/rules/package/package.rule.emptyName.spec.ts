import 'jest';
import Rule from './emptyName';

describe('rule', () => {
  let rule: Rule;
  beforeAll(() => {
    rule = new Rule();
  });

  it('throw', () => {
    const result = rule.validate({
      package: {}
    } as any);
    expect(result).toMatchSnapshot();
  });

  it('not throw', () => {
    const result = rule.validate({
      package: {
        name: 'species'
      }
    } as any);
    expect(result).toMatchSnapshot();
  });
});
