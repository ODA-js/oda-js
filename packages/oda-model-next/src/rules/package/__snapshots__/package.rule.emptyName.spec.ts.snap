// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule not throw 1`] = `Array []`;

exports[`rule throw 1`] = `
Array [
  Object {
    "message": "package must be named",
    "result": "error",
  },
]
`;
